apps/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -272,6 +272,12 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
     return packageDescription;
   }
 
+  private _getValidDependencyVersion(dependencyVersion: string): string {
+    return semver.valid(dependencyVersion.split('/').pop()!) ?
+          dependencyVersion.split('/').pop()! :
+          dependencyVersion.split('_')[0]!;
```

Comment:
For the else case, do we need to grab the last element on split('/') and then grab the first element on split('_')?